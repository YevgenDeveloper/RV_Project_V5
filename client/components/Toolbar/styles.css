@import 'client/styles/variables';

.toolbar {
  display: flex;
  flex-flow: row nowrap;
  margin: 0 auto;
}

.button {
  width: 58px;
  margin: 0 1px 0 0;
  border-radius: 0;
  overflow: hidden;
  text-indent: -1000px;

  &:nth-child(2) {
    border-radius: 4px 0 0 4px;
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
    margin-left: 0;
  }

  &.ride {
    width: 150px;
    border-radius: 4px;
    text-indent: 0;
    margin: 0 5px 0 0;
    padding: 0 15px;
    font-size: $font-size-small;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &.on {
    background-color: $color-blue;

    &:hover:enabled {
      background-color: color-mod($color-blue l(48%));
    }

    &:active:enabled {
      background-color: color-mod($color-blue l(44%));
    }

    &::after {
      opacity: 1;
    }

    &.annotation::after {
      background-image: url('client/assets/icn-annotation-white.svg');
    }
    &.comparison::after {
      background-image: url('client/assets/icn-comparison-white.svg');
    }
    &.measurement::after {
      background-image: url('client/assets/icn-measurement-white.svg');
    }
  }

  &.annotation::after {
    background-image: url('client/assets/icn-annotation.svg');
  }

  &.comparison::after {
    background-image: url('client/assets/icn-comparison.svg');
  }

  &.measurement::after {
    background-image: url('client/assets/icn-measurement.svg');
  }

  &.download::after {
    background-image: url('client/assets/icn-download.svg');
  }

  &.bookmark::after {
    background-image: url('client/assets/icn-bookmark-outline.svg');
  }

  &.aliases::after {
    background-image: url('client/assets/icn-bookmark-outline.svg');
  }

  &.bookmark.bookmarked::after {
    background-image: url('client/assets/icn-bookmark.svg');
  }
}
