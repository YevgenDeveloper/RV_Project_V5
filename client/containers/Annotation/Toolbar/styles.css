@import 'client/styles/variables';

.toolbar {
  z-index: 2;
  flex-shrink: 0;
  padding: 10px;
  display: flex;
  flex-flow: row nowrap;
  background-color: black;
}

.button {
  flex-shrink: 0;
  display: block;
  margin: 0 0 0 10px;
  min-width: 100px;
}

.button.cancel {
  margin-left: auto;
}

.group {
  display: flex;
  flex-flow: row nowrap;
  margin-right: 10px;

  .button {
    width: 50px;
    min-width: 50px;
    margin: 0 1px 0 0;
    border-radius: 0;
    overflow: hidden;
    text-indent: -1000px;
  }

  .button:first-child {
    border-radius: 4px 0 0 4px;
  }

  .button:last-child {
    border-radius: 0 4px 4px 0;
  }

  .button.on,
  .button.on:hover,
  .button.on:active {
    background-color: $color-blue;
  }

  .button.on::after {
    opacity: 1;
  }
}

.button.pen::after {
  background-image: url('client/assets/icn-annotation-white.svg');
}

.button.arrow::after {
  background-image: url('client/assets/icn-arrow-white.svg');
}

.button.text::after {
  background-image: url('client/assets/icn-text-white.svg');
}

.button.rectangle::after {
  background-image: url('client/assets/icn-rectangle-white.svg');
}

.button.circle::after {
  background-image: url('client/assets/icn-circle-white.svg');
}

.button.undo::after {
  background-image: url('client/assets/icn-undo-white.svg');
}

.button.redo::after {
  background-image: url('client/assets/icn-redo-white.svg');
}
