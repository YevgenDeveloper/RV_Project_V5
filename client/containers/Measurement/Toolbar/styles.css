@import 'client/styles/variables';

.toolbar {
  z-index: 2;
  flex-shrink: 0;
  padding: 10px;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  background-color: black;
}

.distance {
  font-size: $font-size-default;
  color: white;
}

.button {
  flex-shrink: 0;
  display: block;
  min-width: 100px;
}
