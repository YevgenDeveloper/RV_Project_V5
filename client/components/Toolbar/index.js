// @flow

import React from 'react';
import classNames from 'classnames/bind';

import { views } from 'client/constants';
import { Button } from 'client/components';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  isBookmarked: boolean,
  disabled: boolean,
  currentRide: string,
  currentView?: string,
  changeView: string => void,
  onClickAnnotation: void => void,
  onClickDownload: void => void,
  onClickBookmark: void => void,
  onClickAliases: void => void,
  onClickRide: void => void,
};

const Toolbar = (props: Props) => (
  <div className={styles.toolbar}>
    <Button
      label={`Ride: ${props.currentRide}`}
      className={cx(styles.button, styles.ride)}
      onClick={props.onClickRide}
      disabled={props.disabled}
    />
    <Button
      label="Annotation"
      className={cx(styles.button, styles.annotation, {
        [styles.on]: props.currentView === views.ANNOTATION,
      })}
      onClick={() => props.onClickAnnotation()}
      disabled={props.disabled}
    />
    <Button
      label="Comparison"
      className={cx(styles.button, styles.comparison, {
        [styles.on]: props.currentView === views.COMPARISON,
      })}
      onClick={() => props.changeView(views.COMPARISON)}
      on={props.currentView === views.COMPARISON}
      disabled={props.disabled}
    />
    <Button
      label="Measurement"
      className={cx(styles.button, styles.measurement, {
        [styles.on]: props.currentView === views.MEASUREMENT,
      })}
      onClick={() => props.changeView(views.MEASUREMENT)}
      on={props.currentView === views.MEASUREMENT}
      disabled={props.disabled}
    />
    <Button
      label="Download"
      className={cx(styles.button, styles.download)}
      onClick={() => props.onClickDownload()}
      disabled={props.disabled}
    />
    <Button
      label="Bookmark"
      className={cx(styles.button, styles.bookmark, {
        [styles.bookmarked]: props.isBookmarked,
      })}
      onClick={() => props.onClickBookmark()}
      disabled={props.disabled}
    />
    <Button
      label="Aliases"
      className={cx(styles.button, styles.aliases)}
      onClick={() => props.onClickAliases()}
      disabled={props.disabled}
    />
  </div>
);

Toolbar.defaultProps = {
  currentView: undefined,
};

export default Toolbar;
