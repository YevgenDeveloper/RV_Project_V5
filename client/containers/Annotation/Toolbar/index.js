// @flow

import React from 'react';

import classNames from 'classnames/bind';
import { Button } from 'client/components';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  disabledSave?: boolean,
  disabledUndo?: boolean,
  disabledRedo?: boolean,
  drawTools: { [drawTool: string]: string },
  drawTool: string,
  onChangeTool: ({ drawTool: string }) => void,
  onClickUndo: void => void,
  onClickRedo: void => void,
  onClickClose: void => void,
  onClickSave: void => void,
};

const Toolbar = (props: Props) => (
  <div className={styles.toolbar}>
    <div className={styles.group}>
      <Button
        dark
        label="Undo"
        className={cx(styles.button, styles.undo, {})}
        onClick={() => props.onClickUndo()}
        disabled={props.disabledUndo}
      />
      <Button
        dark
        label="Redo"
        className={cx(styles.button, styles.redo, {})}
        onClick={() => props.onClickRedo()}
        disabled={props.disabledRedo}
      />
    </div>
    <div className={styles.group}>
      {Object.keys(props.drawTools).map(key => (
        <Button
          dark
          key={key}
          label={props.drawTools[key]}
          className={cx(styles.button, styles[props.drawTools[key]], {
            on: props.drawTool === props.drawTools[key],
          })}
          onClick={() => props.onChangeTool({ drawTool: props.drawTools[key] })}
        />
      ))}
    </div>
    <Button
      dark
      label="Cancel"
      className={cx(styles.button, styles.cancel)}
      onClick={() => props.onClickClose()}
    />
    <Button
      blue
      label="Save"
      className={styles.button}
      onClick={() => props.onClickSave()}
      disabled={props.disabledSave}
    />
  </div>
);

Toolbar.defaultProps = {
  disabledSave: false,
  disabledUndo: false,
  disabledRedo: false,
};

export default Toolbar;
