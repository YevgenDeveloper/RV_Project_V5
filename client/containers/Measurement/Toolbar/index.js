// @flow

import React from 'react';

import { Button } from 'client/components';
import styles from './styles.css';

type Props = {
  calibrated?: boolean,
  distance?: number,
  onClickClose: void => void,
};

const Toolbar = (props: Props) => (
  <div className={styles.toolbar}>
    <div className={styles.distance}>
      {props.calibrated && props.distance
        ? `Distance ${props.distance} m`
        : 'Draw the line between two rails'}
    </div>
    <Button
      dark
      label="Close"
      className={styles.button}
      onClick={() => props.onClickClose()}
    />
  </div>
);

Toolbar.defaultProps = {
  calibrated: false,
  distance: 0,
};

export default Toolbar;
