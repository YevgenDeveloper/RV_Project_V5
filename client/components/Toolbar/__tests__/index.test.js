// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import Toolbar from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Toolbar />);

  expect(wrapper).toMatchSnapshot();
});
