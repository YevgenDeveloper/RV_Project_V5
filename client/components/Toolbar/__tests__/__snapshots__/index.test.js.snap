// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toolbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          blue={false}
          className="button ride"
          dark={false}
          disabled={false}
          label="Ride: undefined"
        />,
        <Button
          blue={false}
          className="button annotation"
          dark={false}
          disabled={false}
          label="Annotation"
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button comparison"
          dark={false}
          disabled={false}
          label="Comparison"
          on={false}
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button measurement"
          dark={false}
          disabled={false}
          label="Measurement"
          on={false}
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button download"
          dark={false}
          disabled={false}
          label="Download"
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button bookmark"
          dark={false}
          disabled={false}
          label="Bookmark"
          onClick={[Function]}
        />,
      ],
      "className": "toolbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button ride",
          "dark": false,
          "disabled": false,
          "label": "Ride: undefined",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button annotation",
          "dark": false,
          "disabled": false,
          "label": "Annotation",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button comparison",
          "dark": false,
          "disabled": false,
          "label": "Comparison",
          "on": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button measurement",
          "dark": false,
          "disabled": false,
          "label": "Measurement",
          "on": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button download",
          "dark": false,
          "disabled": false,
          "label": "Download",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button bookmark",
          "dark": false,
          "disabled": false,
          "label": "Bookmark",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            blue={false}
            className="button ride"
            dark={false}
            disabled={false}
            label="Ride: undefined"
          />,
          <Button
            blue={false}
            className="button annotation"
            dark={false}
            disabled={false}
            label="Annotation"
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button comparison"
            dark={false}
            disabled={false}
            label="Comparison"
            on={false}
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button measurement"
            dark={false}
            disabled={false}
            label="Measurement"
            on={false}
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button download"
            dark={false}
            disabled={false}
            label="Download"
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button bookmark"
            dark={false}
            disabled={false}
            label="Bookmark"
            onClick={[Function]}
          />,
        ],
        "className": "toolbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button ride",
            "dark": false,
            "disabled": false,
            "label": "Ride: undefined",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button annotation",
            "dark": false,
            "disabled": false,
            "label": "Annotation",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button comparison",
            "dark": false,
            "disabled": false,
            "label": "Comparison",
            "on": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button measurement",
            "dark": false,
            "disabled": false,
            "label": "Measurement",
            "on": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button download",
            "dark": false,
            "disabled": false,
            "label": "Download",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button bookmark",
            "dark": false,
            "disabled": false,
            "label": "Bookmark",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
